pinot-common/src/main/java/org/apache/pinot/common/broker/BrokerInfoSelector.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.broker;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+
+public abstract class BrokerInfoSelector implements BrokerSelector {
+
+  protected final boolean _preferTlsPort;
```

Comment:
I am not getting the prefer tls port logic in this PR. IMO, tls vs not is decided somewhere else, and this logic should not worry if the the port is tls. Moreover, the previous code didn't worry about this, so why now? I am missing something here.

Reply:
This is a copy of the current logic. Refer to:

Refer to the following files in the master branch. There are tests as well for this behaviour
* https://github.com/apache/pinot/blob/master/pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/ExternalViewReader.java#L100
* https://github.com/apache/pinot/blob/a6cf197bfa51bc6b5653162d3630c1565ae5b8fa/pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/DynamicBrokerSelector.java#L50
* https://github.com/search?q=repo%3Aapache%2Fpinot%20preferTlsPort&type=code